### DISCUSSION TITLE
How to do Midi Through except a specific type of messages?

### DISCUSSION THREAD

**Discussion by ruiseixasm** [FortySevenEffects/arduino_midi_library#288] [General]
Hi,

I would like to know how to do Midi Through except for a small specific type of messages?

Currently I'm turning off the Midi Thru by doing:

```
// Turn Thru off
MIDI.turnThruOff();
```

However then I have to process all other types of midi messages with Handlers despite just wanting them to pass Thru.

Isn't there a way just to allow pass Thru by type of messages?

Thanks.

**Comment by franky47** [comment]
There is an RFC about this exact use-case in [#40](https://github.com/FortySevenEffects/arduino_midi_library/issues/40#issuecomment-647012093).

Can you review it and let us know there if it would work for you, or if there are other things you'd like to do that aren't covered by the API?

Thanks!
